src/test/kotlin/com/vanniktech/rxriddles/Riddle6Test.kt

```diff
@@ -1,38 +1,41 @@
 package com.vanniktech.rxriddles
 
 import com.vanniktech.rxriddles.solutions.Riddle6Solution
-import com.vanniktech.rxriddles.tools.RxRule
 import io.reactivex.Single
-import org.assertj.core.api.Java6Assertions.assertThat
-import org.junit.Rule
+import org.assertj.core.api.Assertions.assertThat
 import org.junit.Test
-import java.util.concurrent.TimeUnit.SECONDS
-import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit.MILLISECONDS
 
 /** Solution [Riddle6Solution] */
 class Riddle6Test {
-  @get:Rule val rxRule = RxRule()
```

Comment:
I don't use RxRule because I couldn't figure out how to check that the observables were subscribed concurrently but by two different `schedule*` calls. This was necessary to disallow subscribing once for the entire `zip`